### ISSUE TITLE
heap-use-after-free at bvarith64_buffers.c:122

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#236]
Hi, for the following query,
~~~~
(set-logic QF_AUFBV)
(declare-const v0 Bool)
(declare-const v1 Bool)
(declare-const v2 Bool)
(declare-const v3 Bool)
(declare-const v4 Bool)
(declare-const v5 Bool)
(declare-const v6 Bool)
(declare-const v7 Bool)
(declare-const v8 Bool)
(declare-const v9 Bool)
(declare-const v10 Bool)
(declare-const v11 Bool)
(declare-const v12 Bool)
(declare-const v13 Bool)
(declare-const v14 Bool)
(declare-const v15 Bool)
(declare-const v16 Bool)
(declare-const v17 Bool)
(assert true)
(declare-const arr--5847150245411458408_-5847150245408210833-0 (Array (_ BitVec 9) (_ BitVec 12)))
(declare-const v18 Bool)
(declare-const bv_24-0 (_ BitVec 24))
(assert true)
(declare-sort S0 0)
(assert true)
(assert true)
(push 1)
(declare-sort S1 0)
(assert (distinct true true true true true (distinct (store arr--5847150245411458408_-5847150245408210833-0 (bvadd (_ bv276 9) (_ bv276 9)) #x2a4) arr--5847150245411458408_-5847150245408210833-0 arr--5847150245411458408_-5847150245408210833-0) true true true))
(reset-assertions)
(check-sat)
(assert (= (bvadd (_ bv276 9) (_ bv276 9)) (bvadd (_ bv276 9) (_ bv276 9))))
(reset-assertions)
(check-sat)
(declare-const v19 Bool)
(declare-const v20 Bool)
(push 1)
(declare-const v21 Bool)
(declare-const v22 Bool)
(declare-const bv_30-1 (_ BitVec 30))
(declare-const bv_26-0 (_ BitVec 26))
(assert v1)
~~~~

yices2  b63cc30c8c
~~~~
sat
sat
(error "at line 43, column 9: undefined term: v1")
=================================================================
==9219==ERROR: AddressSanitizer: heap-use-after-free on address 0x61b000016468 at pc 0x00000062d619 bp 0x7ffe6fea1de0 sp 0x7ffe6fea1dd0
READ of size 8 at 0x61b000016468 thread T0
    #0 0x62d618 in delete_bvarith64_buffer terms/bvarith64_buffers.c:122
    #1 0x40d1c4 in yices_free_bvarith64_buffer api/yices_api.c:1380
    #2 0x4e79a7 in delete_tstack parser_utils/term_stack2.c:5818
    #3 0x408ce5 in main frontend/yices_smt2.c:766
    #4 0x7f54b71f882f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #5 0x4071a8 in _start (/home/rainoftime/Work/tofuzz/yices2/build/x86_64-pc-linux-gnu-debug/bin/yices_smt2+0x4071a8)

0x61b000016468 is located 1512 bytes inside of 1544-byte region [0x61b000015e80,0x61b000016488)
freed by thread T0 here:
    #0 0x7f54b7dfe32a in __interceptor_free (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x9832a)
    #1 0x706d97 in safe_free utils/memalloc.h:70
    #2 0x70798c in _o_reset_objstore utils/object_stores.c:218
    #3 0x707a7d in reset_objstore utils/object_stores.c:229
    #4 0x69a9c2 in reset_term_manager terms/term_manager.c:365
    #5 0x40cf6a in yices_reset_tables api/yices_api.c:1222
...
~~~~

**Comment by rainoftime** [comment]
~~~~
(set-logic QF_AUFBV)
(declare-const v0 Bool)
(declare-const v1 Bool)
(declare-const v2 Bool)
(declare-const v3 Bool)
(declare-const v4 Bool)
(declare-const v5 Bool)
(declare-const v6 Bool)
(declare-const v7 Bool)
(assert true)
(assert true)
(declare-const arr--2551871311201104581_-3162329498268818509-0 (Array (_ BitVec 10) Bool))
(assert true)
(declare-const arr--2551871311201104581_-6671985692370897704-0 (Array (_ BitVec 10) (Array (_ BitVec 10) Bool)))
(assert true)
(assert true)
(push 1)
(assert true)
(declare-const arr--2551871311194609431_-3428171751036168711-0 (Array (_ BitVec 12) (Array (_ BitVec 10) (Array (_ BitVec 10) Bool))))
(assert true)
(declare-const arr--2551871311194609431_-3162329498268818509-0 (Array (_ BitVec 12) Bool))
(declare-const v8 Bool)
(assert true)
(declare-const arr--2551871311194609431_-3792951240806211522-0 (Array (_ BitVec 12) (Array (_ BitVec 12) Bool)))
(pop 1)
(declare-const v9 Bool)
(declare-const arr--2551871311201104581_-3162329498268818509-3 (Array (_ BitVec 10) Bool))
(assert true)
(declare-const v10 Bool)
(assert true)
(push 1)
(declare-const bv_13-0 (_ BitVec 13))
(declare-const arr--2551871311201104581_-2551871311201104581-0 (Array (_ BitVec 10) (_ BitVec 10)))
(declare-const v11 Bool)
(declare-const arr--2551871311197857006_-2551871311201104581-0 (Array (_ BitVec 9) (_ BitVec 10)))
(assert true)
(assert (distinct true true true true (= (_ bv0 10) (_ bv0 10) (_ bv0 10) (bvsub #b1101101011 #b1101101011) (_ bv0 10))))
(declare-sort S0 0)
(declare-const v12 Bool)
(assert true)
(declare-const v13 Bool)
(declare-const arr--2551871311201104581_-2551871311201104581-3 (Array (_ BitVec 10) (_ BitVec 10)))
(pop 1)
(declare-const v14 Bool)
(declare-const bv_13-0 (_ BitVec 13))
(reset-assertions)
(check-sat)
(assert (= (_ bv0 10) (_ bv0 10) (bvsub #b1101101011 #b1101101011) (bvsub #b1101101011 #b1101101011) (_ bv0 10)))
(declare-const v15 Bool)
(reset-assertions)
(check-sat)
(assert (xor (= bv_13-0 bv_13-0 bv_13-0 bv_13-0 bv_13-0) true true true true true true true true true true))
~~~~

**Comment by BrunoDutertre** [comment]
Same as #237. Fixed by 6c18c3cb6cf10bf63593b7ee35b2eaf936a91e4d.
